pkg/storaged/dialog.jsx

```diff
@@ -724,100 +725,50 @@ export const SelectSpaces = (tag, title, options) => {
-
-                        const on_change = (_event, checked) => {
-                            if (checked && !selected)
-                                change(val.concat(spc));
-                            else if (!checked && selected)
-                                change(val.filter(v => (v != spc)));
-                        };
-
-                        return (
-                            <DataListItem key={spc.block ? spc.block.Device : spc.desc}>
-                                <DataListItemRow>
-                                    <DataListCheck id={(spc.block ? spc.block.Device : spc.desc) + "-row-checkbox"}
-                                                   isChecked={selected} onChange={on_change} />
-                                    <label htmlFor={(spc.block ? spc.block.Device : spc.desc) + "-row-checkbox"}
-                                           className='data-list-row-checkbox-label'>
-                                        <DataListItemCells
-                                            dataListCells={[
-                                                <DataListCell key="select-space-name" className="select-space-name">
-                                                    {format_size_and_text(spc.size, desc)}
-                                                </DataListCell>,
-                                                <DataListCell alignRight isFilled={false} key="select-space-details" className="select-space-details">
-                                                    {block}
-                                                </DataListCell>,
-                                            ]}
-                                        />
-                                    </label>
-                                </DataListItemRow>
-                            </DataListItem>
-                        );
-                    })
-                    }
-                </DataList>
-            );
-        }
-    };
-};
-
-export const SelectSpace = (tag, title, options) => {
-    return {
-        tag,
-        title,
-        options,
-        initial_value: null,
+                <Table variant="compact"
+                       data-field={tag} data-field-type="select-spaces">
+                    <Thead>
+                        <Tr />
+                    </Thead>
+                    <Tbody>
+                        { options.spaces.map((spc, row_index) => {
+                            const selected = (val.indexOf(spc) >= 0);
+                            const block = spc.block ? nice_block_name(spc.block) : "";
+
+                            const on_change = (_event, checked) => {
+                                if (checked && !selected)
+                                    change(val.concat(spc));
+                                else if (!checked && selected)
+                                    change(val.filter(v => (v != spc)));
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18967-20230629-123918-c781551d-fedora-38-devel-storage//Coverage/lcov/github-pr.diff.gcov.html)